### PR TITLE
Support mac wpt builds

### PR DESCRIPTION
**Pull request by sagudev**
There used to be mac wpt runners but it was comment out due to slowness. Now that we have optional workflows it could be useful to have it working but not enabled (available only via `try-wpt-mac` builds)

Currently [failing](https://github.com/sagudev/servo/actions/runs/5224662042/jobs/9434201525) with:
```console
Run python3 ./mach smoketest
Error initializing GStreamer: ErrorLoadingPlugins(["libgstcoreelements.dylib", "libgstnice.dylib", "libgstapp.dylib", "libgstaudioconvert.dylib", "libgstaudioresample.dylib", "libgstgio.dylib", "libgstogg.dylib", "libgstopengl.dylib", "libgstopus.dylib", "libgstplayback.dylib", "libgsttheora.dylib", "libgsttypefindfunctions.dylib", "libgstvolume.dylib", "libgstvorbis.dylib", "libgstaudiofx.dylib", "libgstaudioparsers.dylib", "libgstautodetect.dylib", "libgstdeinterlace.dylib", "libgstid3demux.dylib", "libgstinterleave.dylib", "libgstisomp4.dylib", "libgstmatroska.dylib", "libgstrtp.dylib", "libgstrtpmanager.dylib", "libgstvideofilter.dylib", "libgstvpx.dylib", "libgstaudiobuffersplit.dylib", "libgstdtls.dylib", "libgstid3tag.dylib", "libgstproxy.dylib", "libgstvideoparsersbad.dylib", "libgstwebrtc.dylib", "libgstlibav.dylib", "libgstosxaudio.dylib", "libgstosxvideo.dylib", "libgstapplemedia.dylib", "libgstvideoconvertscale.dylib"])
Servo exited with non-zero status 1
```
even though `./mach bootstrap-gstreamer` is run beforehand.

### CODE REVIEW

**Comment by mrobinson** [comment]
@sagudev is this change ready to review?

**Comment by sagudev** [comment]
I can rebase and retry, but I believe the problem will still persist meaning that bootstrap-gstreamer on mac is not working right.

**Comment by sagudev** [comment]
Run for `try-wpt-mac`: https://github.com/sagudev/servo/actions/runs/5344944881
Run for `try-wpt-mac-2020`: https://github.com/sagudev/servo/actions/runs/5344960898

**Comment by sagudev** [comment]
Still same problems: https://github.com/sagudev/servo/actions/runs/5344960898/jobs/9691405683.

**Comment by mrobinson** [comment]
> Still same problems: https://github.com/sagudev/servo/actions/runs/5344960898/jobs/9691405683.

It looks like the release tarball does not include the GStreamer libraries. Maybe try with this diff:
```diff
diff --git a/.github/workflows/mac.yml b/.github/workflows/mac.yml
index 2724f5044d..49a727ed99 100644
--- a/.github/workflows/mac.yml
+++ b/.github/workflows/mac.yml
@@ -87,7 +87,7 @@ jobs:
           NIGHTLY_REPO_TOKEN: ${{ secrets.NIGHTLY_REPO_TOKEN }}
           NIGHTLY_REPO: ${{ github.repository_owner }}/servo-nightly-builds
       - name: Package binary
-        run: gtar -czf target.tar.gz target/release/servo target/release/*.dylib resources
+        run: gtar -czf target.tar.gz target/release/servo target/release/*.dylib target/release/lib/*.dylib resources
       - name: Archive binary
         uses: actions/upload-artifact@v3
         with:
```

**Comment by sagudev** [comment]
Run for `try-wpt-mac`: https://github.com/sagudev/servo/actions/runs/5358234123
Run for `try-wpt-mac-2020`: https://github.com/sagudev/servo/actions/runs/5358253908

**Comment by sagudev** [comment]
Well the tests are actually running: https://github.com/sagudev/servo/actions/runs/5358253908/jobs/9721262620, but they are really slow ...

NOTE for landing: `wpt-mac-2020` needs to be added to https://github.com/servo/saltfs/blob/master/homu/files/cfg.toml#L30 and new branch created with name `wpt-mac-2020`.

**Comment by jdm** [comment]
Yes, I ran into the issue with significant volumes of timing out tests when I was first porting the buildkite macOS CI to GitHub actions. I was never able to find a solution, despite massively increasing the number of jobs and decreasing parallelism within jobs.

**Comment by sagudev** [comment]
This is much better: https://github.com/sagudev/servo/actions/runs/5769068381/job/15643468800

Only 85 stable unexpected results on layout2020, only half of them TIMEOUT.

**Comment by sagudev** [comment]
Layout2020 has no timeouts for stable unexpected results: https://github.com/sagudev/servo/actions/runs/5769068381/job/15649685095

**Comment by sagudev** [comment]
I think this is ready now as basic functionality is working good enough. This PR intentionally does not allow triggering from try issue commands as it consumes all available mac runners and is still pretty slow. It is primarily meant to be used in personal forks for safety.

We might need OS specific test expectations to make some test pass (UA test for example), but passing tests is outside off this PR.

File: .github/workflows/mac-wpt.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        with:
+          name: wpt-logs-mac-${{ inputs.layout }}
+          path: |
+            test-wpt.${{ matrix.chunk_id }}.log
+            filtered-wpt-results.${{ matrix.chunk_id }}.json
+  report-test-results:
+    name: Reporting test results
+    runs-on: ubuntu-latest
+    if: ${{ always() && !cancelled() }}
+    needs: [mac-wpt]
```
Comment (mrobinson) [inline .github/workflows/mac-wpt.yml:74]: Spaces around the brackets might be nice here as well.

**Review by mrobinson — APPROVED** [review]
Great stuff! I just have a couple nits, but otherwise this looks nice to me.

File: .github/workflows/quick-check.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ name: Quick check
 on:
   push:
     branches:
-      ["**", "!master", "!auto", "!try", "!try-linux", "!try-mac", "!try-windows", "!try-wpt", "!try-wpt-2020", "!dependabot/**"]
+      ["**", "!master", "!auto", "!try", "!try-linux", "!try-mac", "!try-wpt-mac", "!try-wpt-mac-2020", "!try-windows", "!try-wpt", "!try-wpt-2020", "!dependabot/**"]
```
[the remark this answers is not in the document]
Comment (sagudev) [reply .github/workflows/quick-check.yml]: We should probably use branches-ignore.

**Review by mrobinson — APPROVED** [review]

**Pull request merged**